### DISCUSSION TITLE
Dynamic property creation when using BelongsTo

### DISCUSSION THREAD

**Discussion by MartkCz** [cycle/orm#500] [Q&A]
What am I doing wrong?

```php
#[Entity(database: 'root')]
class PortfolioTransaction
{

	#[Column(type: 'string(36)', primary: true, typecast: 'uuid')]
	private UuidInterface $id;

	#[BelongsTo(target: Portfolio::class, nullable: false)]
	private Portfolio $portfolio;
```

```php
#[Entity(database: 'root')]
#[UpdatedAt(field: 'updated')]
class Portfolio
{

	#[Column(type: 'string(36)', primary: true, typecast: 'uuid')]
	private UuidInterface $id;
```

after persisting PortfolioTransaction I get the error `Creation of dynamic property App\Domain\Portfolio\Entity\PortfolioTransaction::$portfolio_id is deprecated`

Value of `portfolio_id` is LazyUuuidFromString object

**Comment by roxblnfk** [comment]
You can define this field in the entity and specify it as `innerKey` in the BelongsTo relation, then the error will not occur.

Cycle uses the `portfolio_id` field to link to the `Portfolio` entity. It seems we don't necessarily need to store this value in the entity, so this mechanism could be avoided in the future.
